File: docs/guides/swift/Sources/mcmSearchWithout.swift
import Foundation
#if os(Linux) // For linux interop
    import FoundationNetworking
#endif

import Core
import Search

func mcmSearchWithout() async throws {
    let getAppIDFor = { (_: String) in "" } // Implement your own logic here
    let getIndexingApiKeyFor = { (_: String) in "" } // Implement your own logic here

    // Fetch from your own data storage and with your own code
    // the associated application ID and API key for this user
    let appID = getAppIDFor("user42")
    let apiKey = getIndexingApiKeyFor("user42")

    let client = try SearchClient(appID: appID, apiKey: apiKey)
    let searchParams = SearchSearchParams.searchSearchParamsObject(
        SearchSearchParamsObject(
            query: "<YOUR_SEARCH_QUERY>",
            facetFilters: .arrayOfSearchFacetFilters([.string("user:user42"), .string("user:public")])
        )
    )

    let response: SearchResponse<Hit> = try await client.searchSingleIndex(
        indexName: "<YOUR_INDEX_NAME>",
        searchParams: searchParams
    )
    print(response)
}

File: docs/guides/csharp/src/mcmSearchWithout.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class McmSearchWithout
{
  private static string GetAppIdFor(string user)
  {
    return ""; // Implement your own logic here
  }

  private static string GetIndexingApiKeyFor(string user)
  {
    return ""; // Implement your own logic here
  }

  async Task Main(string[] args)
  {
    // Fetch from your own data storage and with your own code
    // the associated application ID and API key for this user
    var appId = GetAppIdFor("user42");
    var apiKey = GetIndexingApiKeyFor("user42");

    var client = new SearchClient(new SearchConfig(appId, apiKey));
    var searchParams = new SearchParams(
      new SearchParamsObject
      {
        Query = "<YOUR_SEARCH_QUERY>",
        FacetFilters = new FacetFilters([
          new FacetFilters("user:user42"),
          new FacetFilters("user:public"),
        ]),
      }
    );

    await client.SearchSingleIndexAsync<Hit>("<YOUR_INDEX_NAME>", searchParams);
  }
}
